lib/livebook_web/live/session_live.ex

```diff
@@ -576,15 +585,18 @@ defmodule LivebookWeb.SessionLive do
   end
 
   def handle_params(%{"tab" => tab}, _url, socket) do
+    assert_policy!(socket, :read)
     {:noreply, assign(socket, tab: tab)}
   end
 
   def handle_params(_params, _url, socket) do
+    assert_policy!(socket, :read)
```

Comment:
Some of the routes here may require writing. I think for `handle_params` it may be more robust to define something like `assert_live_action_access!(socket)` where we have an explicit list of all live actions and assert the policy. Then we would just call `assert_live_action_access!(socket)` in every `handle_params` clause.